compiler/noirc_frontend/src/elaborator/patterns.rs

```diff
@@ -581,10 +614,20 @@ impl<'context> Elaborator<'context> {
 
                 generics
             }
-            PathResolutionItem::TraitFunction(_trait_id, Some(generics), _func_id) => {
-                // TODO: https://github.com/noir-lang/noir/issues/6310
-                self.push_err(TypeCheckError::UnsupportedTurbofishUsage { span: generics.span });
-                Vec::new()
+            PathResolutionItem::TraitFunction(trait_id, Some(generics), _func_id) => {
+                let trait_ = self.interner.get_trait(trait_id);
+                let trait_generics = vecmap(&trait_.generics, |generic| {
+                    self.interner.next_type_variable_with_kind(generic.kind())
+                });
+                let kinds = vecmap(&trait_.generics, |generic| generic.kind());
```

Comment:
```suggestion
                let kinds = vecmap(&trait_.generics, |generic| generic.kind());
                let trait_generics = vecmap(kinds, self.interner.next_type_variable_with_kind);                
```
Not sure it compiles, just thought it would make it easier to see that the generics come from the kinds.

Reply:
It didn't work with that syntax, but I pushed a change for that.